lib/datadog/tracing/contrib/mysql2/instrumentation.rb

```diff
@@ -53,12 +53,17 @@ def query(sql, options = {})
                 span.set_tag(Tracing::Metadata::Ext::NET::TAG_TARGET_HOST, query_options[:host])
                 span.set_tag(Tracing::Metadata::Ext::NET::TAG_TARGET_PORT, query_options[:port])
 
+                Contrib::SpanAttributeSchema.set_peer_service!(span, Ext::PEER_SERVICE_SOURCES)
```

Comment:
Do we have test for this?

Reply:
Several, peer.service is a completely independent feature already thoroughly fleshed out. 

We test the function [here](https://github.com/DataDog/dd-trace-rb/blob/master/spec/datadog/tracing/contrib/span_attribute_schema_spec.rb)

We test specifically `mysql2` [here](https://github.com/DataDog/dd-trace-rb/blob/115d4a8c30492e6c9e67ddd2b0620e7cc8747cf3/spec/datadog/tracing/contrib/mysql2/patcher_spec.rb#L102) and we also test more [here](https://github.com/DataDog/dd-trace-rb/blob/115d4a8c30492e6c9e67ddd2b0620e7cc8747cf3/spec/datadog/tracing/contrib/mysql2/patcher_spec.rb#L76) and several places so no worries about that

Reply:
If we already have the `peer.service` tag, why do we need to add this line? Isn't it already in place?

Reply:
It's not being added, it's just being moved to calculate `peer.service` before we do propagation so a value exists within the span to check. Previously we made the last step of every span to calculate `peer.service` but in this case, sql propagation requires `peer.service` to be calculated first and not vice versa